### PR TITLE
Noref update hold request documentation

### PR DESCRIPTION
**Pull request by nonword**
Update hold-request pipeline documentation to include the branching on-site requests logic and update diagram

### CODE REVIEW

File: workflows/patron_hold_request.md [hunk 1]
```diff
@@ -10,21 +10,27 @@ Due to both this approach, and the circumstances under which the LSP was built,
 
 # Hold Request Process Summary
 
-1. Starting in the top-left corner, there are two ways that hold requests can be generated on behalf of NYPL patrons. The standard way is for a hold request button to be pressed in the NYPL Shared Collection Catalog (https://nypl.org/scc) by a logged-in NYPL patron. The other way if for a SCSB user to place the hold in the SCSB UI on behalf of a specific patron. SCSB users can also place multiple requests in the SCSB UI, in which case the requests are split into individual requests before hitting API Gateway.
+1. Starting in the top-left corner, there are two ways that hold requests can be generated on behalf of NYPL patrons. The standard way is for a hold request button to be pressed in the NYPL Shared Collection Catalog (https://nypl.org/rc) by a logged-in NYPL patron. The other way is for a SCSB user to place the hold in the SCSB UI on behalf of a specific patron.
```
Comment (charmingduchess) [inline workflows/patron_hold_request.md:13]: I think it should say "NYPL Research Catalog"

File: workflows/patron_hold_request.md [hunk 2]
```diff
@@ -10,21 +10,27 @@ Due to both this approach, and the circumstances under which the LSP was built,
...
 
-2. If the request originates from Discovery UI, the request will include the patron id fetched from the user’s cookie. If the request originates from SCSB, SCSB will make a request to the patron service to get the patron id from the barcode, before sending that request forward to the HoldRequestService.
+2. If the request originates from RC, the request will include the patron id fetched from the user’s cookie. If the request originates from SCSB, SCSB will make a request to the patron service to get the patron id from the barcode, before sending that request forward to the HoldRequestService.
```
Comment (charmingduchess) [inline workflows/patron_hold_request.md:15]: can you spell this out?

File: workflows/patron_hold_request.md [hunk 3]
```diff
@@ -10,21 +10,27 @@ Due to both this approach, and the circumstances under which the LSP was built,
...
+   * The OnSiteHoldRequestService handles phys and edd requests for on-site items. When the incoming hold is a physical request, it uses the Sierra API to create the hold on the patron's account, and returns the response to the caller. If the incoming request is EDD, it generates a LibAnswers email with information about the request.
+
+8. If the request is for an off-site item, the HoldRequestConsumer will forward the request to SCSB. Critiera for pushing the hold request to SCSB:
+   - The request is for an off-site item (i.e. an item whose location id matches `/^rc/`)
+   - The hold originated in RC (i.e. has a `pickupLocation` or is EDD). If the hold originated in SCSB, their system already knows about it so we do not need to tell them about it through the API
+
+9. Finally, if that request was either a successful EDD request, a successful on-site request, or was /unsuccessful/, (three conditions where there is nothing more to process downstream) HoldRequestConsumer pushes the result to the HoldRequestResult event stream.
    - On the SCSB side, when a request is received:
```
Comment (charmingduchess) [inline workflows/patron_hold_request.md:34]: I think this explanation of what happens in SCSB belongs under step 8.

File: workflows/patron_hold_request.md [hunk 4]
```diff
@@ -10,21 +10,27 @@ Due to both this approach, and the circumstances under which the LSP was built,
...
+   * The OnSiteHoldRequestService handles phys and edd requests for on-site items. When the incoming hold is a physical request, it uses the Sierra API to create the hold on the patron's account, and returns the response to the caller. If the incoming request is EDD, it generates a LibAnswers email with information about the request.
+
+8. If the request is for an off-site item, the HoldRequestConsumer will forward the request to SCSB. Critiera for pushing the hold request to SCSB:
+   - The request is for an off-site item (i.e. an item whose location id matches `/^rc/`)
+   - The hold originated in RC (i.e. has a `pickupLocation` or is EDD). If the hold originated in SCSB, their system already knows about it so we do not need to tell them about it through the API
+
+9. Finally, if that request was either a successful EDD request, a successful on-site request, or was /unsuccessful/, (three conditions where there is nothing more to process downstream) HoldRequestConsumer pushes the result to the HoldRequestResult event stream.
```
Comment (charmingduchess) [inline workflows/patron_hold_request.md:33]: can you specify what the other scenarios are that would not end up getting pushed downstream?

**Review by charmingduchess — CHANGES_REQUESTED** [review]
Thanks for updating this. Just a couple notes.

File: workflows/patron_hold_request.md (hunk 2 above)
Comment (nonword) [reply workflows/patron_hold_request.md:15]: Done (earlier)

File: workflows/patron_hold_request.md (hunk 3 above)
Comment (nonword) [reply workflows/patron_hold_request.md:32]: Whoops. Thanks for reading closely. I mis git-merged. Fixed

**Review by charmingduchess — APPROVED** [review]

**Pull request merged**